src/utils/DependsOnManager.ts

```diff
@@ -1,82 +1,111 @@
-import { Utils } from './Utils';
-import { QueryStateModel, QueryEvents, Component, $$ } from '../Core';
-import { ComponentEvents } from '../ui/Base/Component';
+import { isFunction } from 'underscore';
+import { QueryStateModel, QueryEvents, Component } from '../Core';
 import { MODEL_EVENTS } from '../models/Model';
 import { ComponentsTypes } from './ComponentsTypes';
+import { $$ } from './Dom';
 
 export interface IDependentFacet {
   reset: () => void;
-  toggleDependentFacet: (dependentFacet: Component) => void;
-  element: HTMLElement;
-  root: HTMLElement;
-  dependsOn: string;
-  id: string;
-  queryStateModel: QueryStateModel;
-  bind: ComponentEvents;
+  ref: Component;
+}
+
+export interface IDependentFacetCondition {
+  (facet: Component): boolean;
 }
 
 export class DependsOnManager {
+  private parentFacet: Component;
```

Comment:
I'm wondering if, instead of passing the very generic "ComponentClass", we could instead define a more precise/more targeted interface that all DNE facets implements.

I don't really like what we have here (the base Component class), because it does not tell anything about "what" the `DependsOnManager` needs from it's injected facet dependency, or which contract the facet must fulfill to work correctly inside the manager.

For example, we could have something that says that the injected facet must have a `dependsOn` option property of a precise type. So we could remove the checks that exists currently for this in this class.

Reply:
Yes thats what I discussed during the scrum, the only things we assume are its options though (and there's fallback warnings), and it must work for all facet, not just DNE ones

Reply:
No its just a reference to the parent component, I can remove it and get it directly from the ComponentTypes everytime